### PR TITLE
Latests Posts: Used ToggleGroupControl instead for Image alignment

### LINKED ISSUE #64350: Latests Posts block: "Image alignment" control should use ToggleGroupControl
## What problem does this address?

The [current implementation](https://github.com/WordPress/gutenberg/blob/4b25473b743ab225dbe59ce10341b8397a06a577/packages/block-library/src/latest-posts/edit.js#L310-L319) is hackily repurposing the control intended for the block toolbar, which has the wrong styles, semantics, labeling, and keyboard navigation for something in the block inspector.

Latest Posts block inspector

## What is your proposed solution?

Refactor to use a ToggleGroupControl.

### PR DESCRIPTION
**Pull request by amitraj2203**
## What?
Fixes: #64350 

## Why?
See this [comment](https://github.com/WordPress/gutenberg/issues/64350#issue-2454115382)

## How?
Used ToggleGroupControl for Image alignment

## Testing Instructions
1. Create a post
2. Add `Latest Posts block`
3. Turn on the toggle for `Display featured image`
4. See the `Image alignment` settings

## Screenshots or screencast 
image

### CODE REVIEW

File: packages/block-library/src/latest-posts/edit.js [hunk 1]
```diff
@@ -303,21 +303,35 @@ export default function LatestPostsEdit( { attributes, setAttributes } ) {
...
-								value={ featuredImageAlign }
-								onChange={ ( value ) =>
-									setAttributes( {
-										featuredImageAlign: value,
-									} )
-								}
-								controls={ [ 'left', 'center', 'right' ] }
-								isCollapsed={ false }
+						<ToggleGroupControl
+							className="editor-latest-posts-image-alignment-control"
```
Comment (mirka) [inline packages/block-library/src/latest-posts/edit.js:338]: I guess we should leave this class for back compat, but we can remove the associated CSS in `editor.scss`.

File: packages/block-library/src/latest-posts/edit.js [hunk 2]
```diff
@@ -303,21 +303,35 @@ export default function LatestPostsEdit( { attributes, setAttributes } ) {
...
+								value="left"
+								label={ __( 'Left' ) }
+							/>
+							<ToggleGroupControlOption
+								value="center"
+								label={ __( 'Center' ) }
+							/>
+							<ToggleGroupControlOption
+								value="right"
+								label={ __( 'Right' ) }
```
Comment (mirka) [inline packages/block-library/src/latest-posts/edit.js]: Shouldn't these be `ToggleGroupControlOptionIcon` with the original icons? I guess we could change it to text options if that's better, but that's a separate discussion.

**Review by mirka — COMMENTED** [review]
Thank you for the super quick fix 😄 

I'm not very familiar with how the `block.json` works, but the integration test failure looks relevant. I guess it was previously setting `undefined` or something in the `setAttributes`?

File: packages/block-library/src/latest-posts/block.json [hunk 3]
```diff
@@ -63,7 +63,7 @@
 		},
 		"featuredImageAlign": {
 			"type": "string",
-			"enum": [ "left", "center", "right" ]
+			"default": "none"
```
Comment (t-hamano) [inline packages/block-library/src/latest-posts/block.json]: As you can see from the E2E test failure, updating the block.json definition may affect existing blocks. We should be able to replace it with `ToggleGroupControl` without updating block.json. That is, an `undefined` value is considered to be `'none'`.

Example:

```javascript
<ToggleGroupControl
	label={ __( 'Image alignment' ) }
	value={ featuredImageAlign || 'none' }
	onChange={ ( value ) =>
		setAttributes( {
			featuredImageAlign: value !== 'none' ? value : undefined,
		} )
	}
	isBlock
>

</ToggleGroupControl>
```

File: packages/block-library/src/latest-posts/edit.js (hunk 2 above)
Comment (t-hamano) [reply packages/block-library/src/latest-posts/edit.js]: I agree to keep the original icon. Also, using the `map()` function might make it more readable ([Example](https://github.com/WordPress/gutenberg/blob/720522f3a86f78d5f34e8f364abda370054b6c27/packages/block-editor/src/layouts/constrained.js#L133-L144)).

Comment (amitraj2203) [reply packages/block-library/src/latest-posts/edit.js]: Thanks for the feedback. I have addressed this.

**Comment by t-hamano** [comment]
Thanks for the Update!

I overlooked it, but I think we need to use the "position" alignment icon (`positionLeft`, `positionCenter`, `positionRight`) rather than the "text" alignment icon:

| trunk | This PR |
|--------|--------|
| image | image | 

Additionally, you might want to add `__next40pxDefaultSize` prop to the `ToggleGroupControl`.

**Comment by amitraj2203** [comment]
> Thanks for the Update!
> 
> I overlooked it, but I think we need to use the "position" alignment icon (`positionLeft`, `positionCenter`, `positionRight`) rather than the "text" alignment icon:
> 
> trunk	This PR
> image	image
> Additionally, you might want to add `__next40pxDefaultSize` prop to the `ToggleGroupControl`.

Thanks. Updated the PR.

**Review by t-hamano — APPROVED** [review]
LGTM!

Just to be sure, I would be happy if @mirka could give us some feedback on whether we're using the component itself correctly 🙂

File: packages/block-library/src/latest-posts/edit.js [hunk 4]
```diff
@@ -303,21 +334,33 @@ export default function LatestPostsEdit( { attributes, setAttributes } ) {
...
+							__next40pxDefaultSize
+							label={ __( 'Image alignment' ) }
+							value={ featuredImageAlign || 'none' }
+							onChange={ ( value ) =>
+								setAttributes( {
+									featuredImageAlign:
+										value !== 'none' ? value : undefined,
+								} )
+							}
+							isBlock
```
Comment (mirka) [inline packages/block-library/src/latest-posts/edit.js]: This is just a style guide thing, but for icon-based toggle groups we want to remove the `isBlock` prop.

```suggestion
```

**Review by mirka — APPROVED** [review]
I just had a note about `isBlock`, but everything else looks nice and clean. Thank you! 🚀

File: packages/block-library/src/latest-posts/edit.js (hunk 4 above)
Comment (amitraj2203) [reply packages/block-library/src/latest-posts/edit.js]: Thanks for the info. I have removed it now.

**Comment by t-hamano** [comment]
LGTM! Let's 🚀


https://github.com/user-attachments/assets/94ba6f69-52d5-4112-a860-a3c3a425461f

**Pull request merged**